### PR TITLE
edited Readme

### PR DESCRIPTION
**Pull request by Sergei2404**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,52 @@
-# Hapi
+# HAPI Protocol
 
-This is a crate that helps implement the use of the [HAPI protocol](https://hapi-one.gitbook.io/hapi-protocol/) in your smart contract on Near blockchain.
+[HAPI] is a one-of-a-kind decentralized security protocol that prevents and interrupts any potential malicious activity within the blockchain space. HAPI works by leveraging both external and off-chain data as well as on-chain data accrued directly by HAPI and is publicly available.
```
Comment (ars9) [inline README.md]: Please replace all "HAPI" occurences with "HAPI Protocol" to avoid ambiguity.

Comment (ars9) [reply README.md]: Note: letter capitalization is important.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
 
-The crate has an AML structure that stores:
+# HAPI NEAR Proxy
 
-- account_id: AccountId - the address of the [aml contract](https://github.com/HAPIprotocol/near-proxy-contract);
+It is a proxy [contract](https://github.com/HAPIprotocol/near-proxy-contract) used for reporting and receiving data from [HAPI] protocol main contract on the NEAR blockchain.
```
Comment (ars9) [inline README.md]: Let's replace "reporting and receiving data" with "replicating data".

File: README.md [hunk 3]
```diff
@@ -1,15 +1,52 @@
...
+```
+Address of the contract on the **mainnet**
+
+```
+proxy.hapiprotocol.near
+```
+
+# HAPI NEAR Connector
+
+This crate helps implement the use of the [HAPI] in your smart contract on the NEAR blockchain. You can use it to make request data from the proxy contract, and subsequent processing this data.
```
Comment (ars9) [inline README.md]: "This crate helps to implement HAPI Protocol in your smart contract..." would sound a bit cleaner.

Comment (ars9) [reply README.md]: "to make request" - here "make" isn't needed.

File: README.md [hunk 4]
```diff
@@ -1,15 +1,52 @@
...
+
+# Usage
+
+### You need
+
+1. Add [hapi-near-connector](https://crates.io/crates/hapi-near-connector) to dependencies of your project in Cargo.toml
+
+2. Add use hapi in your lib.rs file
```
Comment (ars9) [inline README.md]: What does "use hapi" mean? Shouldn't it be `use hapi_near_connector`?

File: README.md [hunk 5]
```diff
@@ -1,15 +1,52 @@
...
+
+3. Add field with AML struct to your Contract struct
+
+4. Add `AML::new()` to init of your contract
+
+5. Add cross-contract call to the method on which we need to use hapi
```
Comment (ars9) [inline README.md]: "we need to use the protocol" would look better.

File: README.md [hunk 6]
```diff
@@ -1,15 +1,52 @@
...
+
+6. Create a trait with a callback to handle response from HAPI near proxy.
```
Comment (ars9) [inline README.md]: Please use a proper capitalization of "NEAR".

File: README.md [hunk 7]
```diff
@@ -1,15 +1,52 @@
...
+
+It is short description of steps you need to do to implement hapi in you contract. For detailed information go to [example](#example-of-integration-into-the-ft-contract)
```
Comment (ars9) [inline README.md]: "steps you need to implement HAPI Protocol"

File: README.md [hunk 8]
```diff
@@ -1,15 +1,52 @@
...
+
+### The crate has an AML structure that stores
+
+- account_id: AccountId - the address of the HAPI proxy [contract](https://github.com/HAPIprotocol/near-proxy-contract);
 - pub aml_conditions: UnorderedMap<Category, RiskScore> - a map of categories and corresponding risk levels that you will add.
 
 >Note
 >
->If the risk level was not set for some categories, then the risk level for the category **All** is used, which is set during initialization.
+>If the risk level was not set for some categories, then the risk level for the category **All** is used.
```
Comment (ars9) [inline README.md:49]: Can you describe the configuration for a case when the consumer doesn't need to differentiate between categories at all?

File: README.md [hunk 9]
```diff
@@ -22,8 +59,144 @@ ___________
...
+hapi-near-connector = "0.3.0"
+```
+
+2. Add use hapi in your lib.rs file
+```
+use hapi_near_connector::aml::*;
+```
+
+3. Add AML struct to your Contract struct
+![img1](img/img1.png)
```
Comment (ars9) [inline README.md]: This shouldn't be an image.

File: README.md [hunk 10]
```diff
@@ -22,8 +59,144 @@ ___________
...
+
+4. Add "AML new" to init of your contract
+
+>Note. Here we set the accepted risk level as MAX_RISK_LEVEL/2 i.e 10/2 = 5. 
+```
+aml: AML::new(aml_account_id, MAX_RISK_LEVEL/2)
+```
+
+5. Add cross-contract call to the method on which we need to use hapi. 
+* *ext_aml* - it's hapi's trait.
```
Comment (ars9) [inline README.md]: "it's the connector's trait"

File: README.md [hunk 11]
```diff
@@ -22,8 +59,144 @@ ___________
...
+![img1](img/img1.png)
+
+4. Add "AML new" to init of your contract
+
+>Note. Here we set the accepted risk level as MAX_RISK_LEVEL/2 i.e 10/2 = 5. 
+```
+aml: AML::new(aml_account_id, MAX_RISK_LEVEL/2)
+```
+
+5. Add cross-contract call to the method on which we need to use hapi.
```
Comment (ars9) [inline README.md]: "to use the connector".

File: README.md [hunk 12]
```diff
@@ -22,8 +59,144 @@ ___________
...
+```rust
+fn ft_transfer(
+        &mut self,
+        receiver_id: AccountId,
+        amount: U128,
+        memo: Option<String>,
+) -> Promise {
+    assert_one_yocto();
+    let sender: AccountId = predecessor_account_id();
+    ext_aml::ext(self.aml.get_account())
```
Comment (ars9) [inline README.md:130]: Can you please add a short comment that explains what happens in this block depending on the protocol data?

File: README.md [hunk 13]
```diff
@@ -22,8 +59,144 @@ ___________
...
+        sender_id: AccountId,
+        #[callback] category_risk: CategoryRisk,
+        receiver_id: AccountId,
+        amount: U128,
+        memo: Option<String>,
+    );
+}
+```
+
+After that, your contract can already work with Hapi.
```
Comment (ars9) [inline README.md]: "work with HAPI Protocol"

File: README.md [hunk 14]
```diff
@@ -22,8 +59,144 @@ ___________
...
+Also, you can delete an added category. Then it will be evaluated in the *All* category.
+>Note. You can't delete the category *All*.
+
+```rust
+fn remove_category(&mut self, category: Category) {
+    self.assert_owner();
+    self.aml.remove_category(category);
+}
+```
+
```
Comment (ars9) [inline README.md:196]: Let's add a suggestion to configure categories during a migration if that's something the contract owner doesn't plan to do often.

File: README.md [hunk 15]
```diff
@@ -22,8 +59,144 @@ ___________
...
+            metadata: old_contract.metadata,
+            owner_id: old_contract.owner_id,
+            aml: AML::new(aml_account_id, MAX_RISK_LEVEL / 2),
+        }
+    }
+}
+```
+# Alternative
+
+Also you can integrate HAPI without crate, you can find [example on Jumbo exchange](https://github.com/jumbo-exchange/contracts#hapi-protocol-integration).
```
Comment (ars9) [inline README.md]: "integrate HAPI Protocol without this crate"

**Review by ars9 — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (Sergei2404) [reply README.md:49]: For example, when the user agrees that all categories in our list of categories are unacceptable for him.

File: README.md [hunk 16]
```diff
@@ -1,10 +1,10 @@
 # HAPI Protocol
 
-[HAPI] is a one-of-a-kind decentralized security protocol that prevents and interrupts any potential malicious activity within the blockchain space. HAPI works by leveraging both external and off-chain data as well as on-chain data accrued directly by HAPI and is publicly available.
+[HAPI Protocol] is a one-of-a-kind decentralized security protocol that prevents and interrupts any potential malicious activity within the blockchain space. HAPI Protocol works by leveraging both external and off-chain data as well as on-chain data accrued directly by HAPI Protocol and is publicly available.
 
 # HAPI NEAR Proxy
 
-It is a proxy [contract](https://github.com/HAPIprotocol/near-proxy-contract) used for reporting and receiving data from [HAPI] protocol main contract on the NEAR blockchain.
+It is a proxy [contract](https://github.com/HAPIprotocol/near-proxy-contract) used for replicating data from [HAPI Protocol] protocol main contract on the NEAR blockchain.
```
Comment (ars9) [inline README.md]: Please remove the duplicate "protocol".

File: README.md [hunk 17]
```diff
@@ -139,9 +152,27 @@ pub trait ExtContract {
...
+        // here we check response from proxy contract, if you don't need to assert use "check_risk" function
+        self.aml.assert_risk(category_risk);
+        // and then you can to continue performing the function
+        self.token.ft_transfer(sender_id, receiver_id, amount, memo)
+    }
+}
 ```
 
-After that, your contract can already work with Hapi.
+After that, your contract can already work with HAPI Protocol.
```
Comment (ars9) [inline README.md]: "your contract is ready to work with"

File: README.md [hunk 18]
```diff
@@ -185,18 +216,23 @@ impl Migrations for Contract {
 
         let old_contract: OldContract = env::state_read().expect("Old state doesn't exist");
 
+        let mut aml = AML::new(aml_account_id, MAX_RISK_LEVEL / 2);
+
+        // if you doesn't plan to add categories often you can do it right away
```
Comment (ars9) [inline README.md]: "if you don't plan to modify categories"

**Review by ars9 — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (ars9) [reply README.md:49]: I mean add it to this note :)

**Review by ars9 — APPROVED** [review]
LGTM

**Pull request merged**